### PR TITLE
Do not batch reduce as keras optimzer already does.

### PR DESCRIPTION
**Pull request by acmore**
# Description

Brief Description of the PR:

This pr is to remove `batch_reduce_to` call on the gradients for optimizer v2. In the [optimzer v2 base class](https://github.com/tensorflow/tensorflow/blob/919a858f97dfdb219ea3d1a084b594bfa520ef59/tensorflow/python/keras/optimizer_v2/optimizer_v2.py#L668), the gradients are aggregated already. 

## Type of change

- [x] Bug fix
- [ ] New Tutorial
- [ ] Updated or additional documentation
- [ ] Additional Testing
- [ ] New Feature

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/recommenders-addons/blob/master/CONTRIBUTING.md#coding-style)
    - [x] By running yapf
    - [x] By running clang-format
- [ ] This PR addresses an already submitted issue for TensorFlow Recommenders-Addons
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*

### CODE REVIEW

**Comment by MoFHeka** [comment]
Did you check for tf.compat.v1.train.optimzer?

**Comment by acmore** [comment]
> tf.compat.v1.train.optimzer

Correct me if I am wrong. I think this `_distributed_apply` was replaced only in keras optimizer?

**Comment by MoFHeka** [comment]
> > tf.compat.v1.train.optimzer
> 
> Correct me if I am wrong. I think this `_distributed_apply` was replaced only in keras optimizer?

**Comment by MoFHeka** [comment]
> > tf.compat.v1.train.optimzer
> 
> Correct me if I am wrong. I think this `_distributed_apply` was replaced only in keras optimizer?

Sorry, you're right. And this bug seems may cause a problem if we don't assign specific local device when using eras model.

**Comment by acmore** [comment]
> > > tf.compat.v1.train.optimzer
> > 
> > 
> > Correct me if I am wrong. I think this `_distributed_apply` was replaced only in keras optimizer?
> 
> Sorry, you're right. And this bug seems may cause a problem if we don't assign specific local device when using eras model. Please run "find ./ -name *.py -exec yapf --style=./.yapf -ir {} ;" for formatting.

done

**Comment by Lifann** [comment]
It does call batch_reduce_to twice on one call. LGTM.

**Review by Lifann — APPROVED** [review]
LGTM

**Review by MoFHeka — APPROVED** [review]
LGTM

**Review by Lifann — DISMISSED** [review]

**Review by rhdong — APPROVED** [review]
LGTM

**Comment by rhdong** [comment]
Hi @acmore, thank you for the fix, and please squash the 3 commits into one, after that I will merge it.

**Comment by acmore** [comment]
> Hi @acmore, thank you for the fix, and please squash the 3 commits into one, after that I will merge it.

Thanks. I have squashed them into one.
